libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1210,6 +1233,13 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 				"Tried to use .push() on a non-dynamically sized array"
 			);
 		}
+		else if (member == "pop")
+		{
+			solAssert(
+				type.isDynamicallySized() && type.location() == DataLocation::Storage,
```

Comment:
Should probably also check whether `type.category()` is `Array`.

Reply:
I think it should be also safe without, because we're switching on ``_memberAccess.expression().annotation().type->category()`` to reach this branch. But it  doesn't hurt to have it checked since one could mess things up in between (function is rather longish :)).